consai_examples/consai_examples/field_observer.py

```diff
@@ -764,3 +764,56 @@ def _forward_robots_id(self, my_robot_id, my_robot_pos, robots_pos, robots_vel,
 
         return forward_robots_id
 
+    # シューターが相手ゴールに向かってシュートできるかを判定する関数
+    def get_shoot_point(self, my_robot_id):
+        # ゴールポストのx座標
+        goal_post_x = 6
+        # ゴールポストのy座標
+        goal_post_y = 0.9
+        # フィールド中央から見たゴールポストの座標
+        goal_post = [[goal_post_x, goal_post_y], [goal_post_x, - goal_post_y]]
+        # シュートを打つ目標点の座標
+        goal_center_target = [goal_post_x, 0]
+        # ロボットの実直径
+        robot_diameter = 0.2
+        # ロボットの想定移動時間（移動しないものとして扱う）
+        dt = 0
+
+        # 味方ロボットの位置、相手ロボットの位置と速度を取得
+        our_robots_pos = self.our_robots_pos
+        their_robots_pos = self.their_robots_pos
+        their_robots_vel = self.their_robots_vel
+
+        my_robot_pos = our_robots_pos[my_robot_id]
+
+        # ボールを持っている味方ロボットより前方にいる相手ロボットのIDをリストに格納する
+        forward_their_robots_id = self._forward_robots_id(my_robot_id, my_robot_pos, their_robots_pos, their_robots_vel, robot_diameter, dt)
+        # シューターの前方に相手ロボットが存在しない場合
+        if len(forward_their_robots_id) == 0:
+            # 相手ゴールの中央の座標を返す
+            return goal_center_target
+
+        for robot_id in forward_their_robots_id:
+            # パサーとレシーバー候補ロボットを結ぶ直線の傾きと切片を取得
+            slope_from_shooter_to_goal, intercept_from_shooter_to_goal = tool.get_line_parameter(their_robots_pos[robot_id], our_robots_pos[my_robot_id])
+
+            # シューターから妨害する相手ロボットとの直線と相手ゴールのラインとの交点算出
+            edge_of_goal_possible_area = slope_from_shooter_to_goal * goal_post_x + intercept_from_shooter_to_goal
+
+            # ゴールポストに近い側をゴールできる範囲の端に設定
+            if abs(goal_post[0][1] - edge_of_goal_possible_area) < abs(goal_post[1][1] - edge_of_goal_possible_area) and abs(edge_of_goal_possible_area) < goal_post_y:
+                goal_post[0][1] = edge_of_goal_possible_area
+            elif abs(goal_post[0][1] - edge_of_goal_possible_area) > abs(goal_post[1][1] - edge_of_goal_possible_area) and abs(edge_of_goal_possible_area) < goal_post_y:
+                goal_post[1][1] = edge_of_goal_possible_area
+
+        # シュート可能なゴールの幅
+        goal_width = goal_post[0][1] - goal_post[1][1]
+
+        # ロボット2台分の直径よりもゴールできる幅が広いときはその範囲の中心座標を返す
+        if goal_width > robot_diameter * 2:
+            goal_center_target[1] = goal_width / 2 - goal_post_y
+        # ロボットの直径よりもゴールできる幅が小さければ空のリストを返す
+        else:
+            goal_center_target = []
+        return goal_center_target
+    
```

Comment:
goal_center_targetは定数なので書き換えるのはやめましょう（goal_center_targetだと思ったら空リストでした、となる）


```suggestion
        target_pos = goal_center_target
        if goal_width > robot_diameter * 2:
            target_pos[1] = goal_width / 2 - goal_post_y
        # ロボットの直径よりもゴールできる幅が小さければ空のリストを返す
        else:
            target_pos  = []
        return target_pos
    
```